docs/DeveloperGuide.md

```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 (For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC11 View all appointments within a timeframe**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to view all appointments within a timeframe. 
+2. MediConnect shows the appointments list.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The command is invalid.
+    * 1a1. MediConnect shows an error message.
+
+      Use case resumes at step 1.
+
+
+**Use case: UC12 View a patient’s upcoming appointments**
+
+**MSS**
+
+1. User requests to view a patient’s upcoming appointments 
+2. MediConnect shows the patient’s appointment list
+
+   use case ends.
+
+**Extensions**
+
+* 1a. The command is invalid.
+    * 1a1. MediConnect shows an error message.
+
+      Use case resumes at step 1.
+
+* 1b. The patient does not exist in the system.
+    * 1b1. MediConnect shows an error message.
+    
+      Use case resumes at step 1.
+
+
+**Use case: UC13 View a doctor’s upcoming appointments**
+
+**MSS**
+
+1. User requests to view a doctor’s upcoming appointments. 
+2. MediConnect shows the doctor’s appointment list.
+
+   use case ends.
+
+**Extensions**
+
+* 1a. The command is invalid.
+    * 1a1. MediConnect shows an error message.
+
+      Use case resumes at step 1.
+
+* 1b. The doctor does not exist in the system.
+    * 1b1. MediConnect shows an error message.
+
+      Use case resumes at step 1.
+
+
+**Use case: UC14 Add an appointment between a patient and doctor**
+
+**MSS**
+
+1. User requests to add a specific appointment between a patient and doctor. 
+2. MediConnect adds the appointment to the list.
+
+   use case ends.
+
+**Extensions**
+
+* 1a.MediConnect detects that the arguments provided to add an appointment is invalid.
+    * 1a1. MediConnect shows an error message.
+    * 1a2 MediConnect requests for the correct input.
+      Use case resumes at step 1.
+
+* 1b. MediConnect detects that the appointment to be added already exists in the system
+    * 1b1. MediConnect shows an error message.
+
+      Use case ends.
+
+**Use case: UC15 Edit an appointment**
+
+**MSS**
+
+1. User requests to edit an appointment.
+2. MediConnect shows the current details of the appointment to the user.
+3. requests to edit the appointment. 
```

Comment:
missing MediConnect? Or duplicated from step 1? Actually step 2 shouldnt be here also. Probably just request to edit then show the updated information